x-pack/plugins/security_solution/common/detection_engine/utils.ts

```diff
@@ -93,9 +93,16 @@ export const isSuppressionRuleConfiguredWithMissingFields = (ruleType: Type) =>
 export const isSuppressionRuleInGA = (ruleType: Type): boolean => {
   return isSuppressibleAlertRule(ruleType) && SUPPRESSIBLE_ALERT_RULES_GA.includes(ruleType);
 };
-
-export const shouldShowResponseActions = (ruleType: Type | undefined) => {
+export const shouldShowResponseActions = (
+  ruleType: Type | undefined,
+  automatedResponseActionsForAllRulesEnabled: boolean
+) => {
   return (
-    isQueryRule(ruleType) || isEsqlRule(ruleType) || isEqlRule(ruleType) || isNewTermsRule(ruleType)
+    isQueryRule(ruleType) ||
+    isEsqlRule(ruleType) ||
+    isEqlRule(ruleType) ||
+    isNewTermsRule(ruleType) ||
+    (automatedResponseActionsForAllRulesEnabled &&
+      (isThresholdRule(ruleType) || isThreatMatchRule(ruleType) || isMlRule(ruleType)))
```

Comment:
Consider increasing readability by going, for example, this route:
```typescript
export const shouldShowResponseActions = (
  ruleType: Type | undefined,
  automatedResponseActionsForAllRulesEnabled: boolean
): boolean => {
  const isEligibleRuleType = 
    isQueryRule(ruleType) || 
    isEsqlRule(ruleType) || 
    isEqlRule(ruleType) || 
    isNewTermsRule(ruleType);

  const isAutomatedResponseApplicable = 
    automatedResponseActionsForAllRulesEnabled && 
    (isThresholdRule(ruleType) || isThreatMatchRule(ruleType) || isMlRule(ruleType));

  return isEligibleRuleType || isAutomatedResponseApplicable;
};
```

Reply:
Yes, I agree 👍 this would increase readability. However, this check will be removed next week altogether with the feature flag, so I would prefer not to change it now, and just merge the thing :P 
Are you ok with that? :)

Reply:
Sure :)